packages/cli/src/util/get-latest-version/get-latest-worker.js

```diff
@@ -163,12 +163,27 @@ process.once('message', async msg => {
       output.debug('Available dist tags:', Object.keys(tags));
     }
 
+    // if the latest version is newer than the previous latest version, then
+    // invalidate `notifyAt` to force the notification to be displayed,
+    // otherwise keep the the existing `notifyAt`
+    let notifyAt;
+    try {
+      const old = JSON.parse(await readFile(cacheFile, 'utf-8'));
+      notifyAt =
+        old?.version && old.version === version ? old?.notifyAt : undefined;
+    } catch (err) {
+      // cache does not exist or malformed
+      if (err.code !== 'ENOENT') {
+        output.debug(`Error reading latest package cache file: ${err}`);
+      }
+    }
```

Comment:
Normally I would agree, but this is a `.js` file, not a `.ts` file, and this `get-latest-worker.js` is not bundled, but rather gets copied at build time into the dist directory. Because of this and because of past issues on dependencies, `get-latest-worker.js` is designed to not have any non-Node.js dependencies, for better or worse.

Reply:
I was suggesting extracting a function. It can live in this file. I don't we need a dependency here.

Reply:
Ah, yes! I misunderstood, but now I see the 💡!

Not only is is this a good idea, I decided to bust the dist tag fetching out into a separate function too.